### PR TITLE
Add the hasListeners method to client socket.

### PR DESCRIPTION
**Pull request by Pringels**
Fixes: #295

### CODE REVIEW

**Comment by Pringels** [comment]
@thoov apologies for the direct ping, but I would love to get this merged :)

**Comment by Atrue** [comment]
@Pringels What's the use case of having these methods?
Native WebSocket client and server implementation don't have these methods

**Comment by Pringels** [comment]
@Atrue this change simply extends the already existing [limited socket.io support](https://github.com/thoov/mock-socket#socket-io), not Native WebSockets.

File: src/event/target.js [hunk 1]
```diff
@@ -44,6 +44,21 @@ class EventTarget {
     this.listeners[type] = reject(arrayOfListeners, listener => listener === removingListener);
   }
 
+  /*
+   * Checks if a given event type has registered event listeners
+   *
+   * @param {string} type - the type of event (ie: 'open', 'message', etc.)
+   */
+  hasEventListeners(type) {
```
Comment (Atrue) [inline src/event/target.js:52]: please move this logic to `hasListeners` method. Native EventTarget doesn't have it

**Comment by Atrue** [comment]
@Pringels Please use yarn to install packages and make sure you don't have any updated lock files

**Comment by Pringels** [comment]
@Atrue please take another look. I've also updated the `off()` method to match the `Emitter` class as described here: https://socket.io/docs/v4/client-api/#socketoneventname-callback

**Pull request merged**